src/main.c

```diff
@@ -52,6 +66,7 @@ int main(int argc, char **argv)
     MPI_Abort(MPI_COMM_WORLD, EXIT_FAILURE);
     MPI_Finalize();
 #endif
+    perror(NULL);
```

Comment:
Not sure this line should be here - may be above the `#endif`?

Reply:
I think we agreed to not use `perror(NULL)` anywhere else and set it here when the error is handled.  

So whenever a function fails, it just returns the appropriate exit value without having to worry about setting `perror(NULL)`.